### PR TITLE
feat(hz): compatibility annotations for protobuf

### PR DESCRIPTION
**Pull request by FGYFFFF**
#### What type of PR is this?

feat
#### Check the PR title.

- [x] This PR title match the format: `<type>(optional scope): <description>`.
- [x] The description of this PR title is user-oriented and clear enough for others to understand.
- [x] Attach the PR updating the user documentation if the current PR requires user awareness at the usage level. [User docs repo](https://github.com/cloudwego/cloudwego.github.io).

#### (Optional) Translate the PR title into Chinese.
为 protobuf 添加兼容性注解

#### (Optional) More detail description for this PR(en: English/zh: Chinese).

en:  Add compatibility annotations to protobuf to avoid serial number conflicts
zh(optional): 为 protobuf 添加兼容性注解，避免出现序号冲突

#### (Optional) The PR that updates user documentation:

https://github.com/cloudwego/cloudwego.github.io/pull/621/commits

### CODE REVIEW

File: cmd/hz/protobuf/api/api.proto [hunk 1]
```diff
@@ -15,9 +15,14 @@ extend google.protobuf.FieldOptions {
   optional string path = 50106;
   optional string vd = 50107;
   optional string form = 50108;
-  optional string go_tag = 51001;
   optional string js_conv = 50109;
   optional string file_name = 50110;
+
+  // 50131~50160 used to extend field option by hz
+  optional string go_tag = 51001;
```
Comment (welkeyever) [inline cmd/hz/protobuf/api/api.proto:22]: 这个往后面丢？看起来跟注释不符

File: cmd/hz/protobuf/ast.go [hunk 2]
```diff
@@ -284,9 +295,9 @@ func parseAnnotationToClient(clientMethod *generator.ClientMethod, gen *protogen
 			}
 		}
 
-		if proto.HasExtension(f.Desc.Options(), api.E_Form) {
+		if proto.HasExtension(f.Desc.Options(), api.E_Form) || proto.HasExtension(f.Desc.Options(), api.E_FormCompatible) {
```
Comment (welkeyever) [inline cmd/hz/protobuf/ast.go:298]:
```
if f := getCompatibleAnnotation(f.Desc.Options(), api.E_Form, api.E_FormCompatible); f != nil {
    formAnnos := f
    hasAnnotation = true
    ....

}
```

这样是不就行了，可以少一次has 判断？

File: cmd/hz/protobuf/api/api.proto (hunk 1 above)
Comment (FGYFFFF) [reply cmd/hz/protobuf/api/api.proto]: done

File: cmd/hz/testdata/protobuf3/api.proto [hunk 3]
```diff
@@ -29,29 +34,29 @@ extend google.protobuf.MethodOptions {
   optional string options = 50206;
   optional string head = 50207;
   optional string any = 50208;
-  optional string re_get = 50211;
```
Comment (welkeyever) [inline cmd/hz/testdata/protobuf3/api.proto:32]: 这些re_开头的又是啥

File: cmd/hz/protobuf/ast.go (hunk 2 above)
Comment (FGYFFFF) [reply cmd/hz/protobuf/ast.go]: done

**Review by Duslia — APPROVED** [review]

**Review by welkeyever — APPROVED** [review]

**Pull request merged**